data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/MultiBufferDecorator.java

```diff
@@ -42,6 +42,19 @@ public Duration getDrainTimeout() {
                 .reduce(Duration.ZERO, Duration::plus);
     }
 
+    @Override
+    public Integer getMaxRequestSize() {
+        Integer result = null;
+        for (final Buffer buffer : allBuffers) {
+            Integer maxRequestSize = buffer.getMaxRequestSize();
+            if (maxRequestSize != null) {
+                if (result == null || result > maxRequestSize)
+                    result = maxRequestSize;
```

Comment:
Shouldn't we get the minimum max request size?

Also, a stream would probably work nicely here:

```
allBuffers.stream().mapToLong(b -> b.getMaxRequestSize()).min();
```